strict-checked-vars/strict-checked-vars.cabal

```diff
@@ -0,0 +1,75 @@
+cabal-version:   3.0
+name:            strict-checked-vars
+version:         0.1.0.0
+synopsis:
+  Strict MVars and TVars with invariant checking for IO and IOSim
+
+description:
+  Strict @MVar@ and @TVar@ interfaces with invariant checking compatible with
+  [IO](https://hackage.haskell.org/package/base-4.18.0.0/docs/Prelude.html#t:IO)
+  & [io-sim](https://hackage.haskell.org/package/io-sim).
+
+license:         Apache-2.0
+license-files:
+  LICENSE
+  NOTICE
+
+copyright:       2019-2023 Input Output Global Inc (IOG).
+author:          IOG Engineering Team
+maintainer:      operations@iohk.io
+category:        Concurrency
+build-type:      Simple
+extra-doc-files:
+  CHANGELOG.md
+  README.md
+
+bug-reports:     https://github.com/input-output-hk/io-sim/issues
+tested-with:     GHC ==8.10 || ==9.2 || ==9.4 || ==9.6
+
+source-repository head
+  type:     git
+  location: https://github.com/input-output-hk/ouroboros-consensus
+  subdir:   strict-checked-vars
+
+library
+  hs-source-dirs:   src
+  exposed-modules:
+    Control.Concurrent.Class.MonadMVar.Strict.Checked
+    Control.Concurrent.Class.MonadSTM.Strict.TVar.Checked
+
+  default-language: Haskell2010
+  build-depends:
+    , base         >=4.9  && <4.19
+    , io-classes   ^>=1.1
+    , strict-mvar  ^>=1.1
+    , strict-stm   ^>=1.1
+
+  ghc-options:
+    -Wall -Wno-unticked-promoted-constructors -Wcompat
+    -Wincomplete-uni-patterns -Wincomplete-record-updates
+    -Wpartial-fields -Widentities
```

Comment:
I suspect that `-Wno-unticked-promoted-constructors` is not really needed.  In `io-classes` we use these flags:
```
    ghc-options: -Wall
                 -Wcompat
                 -Wincomplete-uni-patterns
                 -Wincomplete-record-updates
                 -Wpartial-fields
                 -Widentities
                 -Wunused-packages
                 -Wno-redundant-constraints
                 -Wno-unticked-promoted-constructors
```